File: IBlogmart-SPA/src/app/_models/category.ts
export interface Category {
    id?: number;
    name: string;
    active: boolean;
    mainImageUrl?: string;
    images?: [ {
        id: number;
        url: string;
        isMain: boolean;
        type?: number;
    }];
    subCategories?: [ {
        id: number;
        name: string;
        active: boolean;
        categoryId: number;
    }];
 }

File: IBlogmart.api/Models/Category.cs
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace IBlogmart.api.Models
{
    public class Category
    {
        [Key]
        public int Id { get; set; }
        public string Name {get;set;}
        public bool Active {get;set;}
        public ICollection<Image> Images {get;set;}
       
    }
}